bookings/models.py

```diff
@@ -100,3 +100,13 @@ def confirm(self, passed_parameters=None):
             self.transaction_id = transaction_id
         self.save()
         return response_data.get("tickets", [])
+
+    def retrieve(self, passed_parameters=None):
```

Comment:
It seems to be quite popular in Django community, but I don't like the idea of having API calls mixed into models. The business logic should be in a service like class.

Reply:
Querysets and models are usually indeed common places to add logic which might get called from various places. I guess we can think about if we would want to refactor the methods which call the ticketing system API out of the booking queryset and model.